design-proposals/migration-policies.md

```diff
@@ -0,0 +1,201 @@
+* Preventing misconfigurations regarding overlapping selectors
+
+## Definition of Users
+This feature's most obvious user is the cluster administrator.
+
+VM owners/creators, by contrast, should not know or care about their migration configuration in most cases.
+Moreover, the design should take into account that one of the goals is to allow the cluster administrator to dynamically
+define and change existing policies with as little participation as possible from the VM owner.
+
+Another goal is to let the admin the possibility to expose different policies to the users in a way he can then
+use. For example, an admin can set two policies for fast/slow migration that will apply to any VM with a
+`migration-type: slow` or `migration-type: fast` and this way a user can apply different policies by defining
+the relevant label.
+
+## User Stories
+As an admin I would like to have different migration policies for different groups of VMs.
+* By "migration policies" I mean controlling one or more of the following migration aspects:
+    * Bandwidth
+    * Scheduling
+    * Storage
+* By “different groups of VMs” I could mean:
+    * All VMs in the same namespace
+    * All VMs that have certain label defined
+    * All VMs that have certain label with certain value defined
+* As an admin I know that some workloads are in higher priority than others and therefore need different 
+  migration policies.
+* As an admin I know that some workloads are not migration-friendly and demand careful configurations in order
+  for migrations to converge.
+* As an admin I want to have the ability to temporarily tune migrations for a specific event such as draining
+  a node or performing an upgrade.
+* As an admin I want that the migration policy would be as transparent as possible for the VM owner / creator.
+* As a user I want to be able to tell admins the workload characteristics of my VMs so that admins can create
+  policies and/or namespaces with the desired characteristics.
+
+## Repos
+Kubevirt/kubevirt
+
+# Design
+The design will be similar to Kubernetes' [NetworkPolicy](https://kubernetes.io/docs/concepts/services-networking/network-policies/)
+which is a similar concept. Therefore, a new MigrationPolicy CRD will be introduced. This CRD's spec will contain all
+relevant fields from KubevirtCR's [MigrationConfiguration](https://kubevirt.io/api-reference/master/definitions.html#_v1_migrationconfiguration)
+(i.e. which are not inherently cluster-wide configurations). In the future, additional
+configurations (or status related fields) that inherently relate to non-global context could be added,
+thus the configurations on NetworkPolicy and KubevirtCR's MigrationConfigurations may diverge to some degree.
+
+In addition, the new CRD's spec will include the following ways of specifying the groups of VMIs on which
+to apply the policy:
+* _(By VMI names / regular expressions)?_
+* By VMI's labels
+* By namespace's name
+* By namespace's labels
```

Comment:
I think we should limit this matching to namespace's labels and VMI's labels for now. If that isn't sufficient, we can take another look at this in the future.

Reply:
Sure, done